oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -123,25 +123,19 @@ protected ExternalAccountCredentials(
       @Nullable String clientId,
       @Nullable String clientSecret,
       @Nullable Collection<String> scopes) {
-    this.transportFactory =
-        MoreObjects.firstNonNull(
-            transportFactory,
-            getFromServiceLoader(HttpTransportFactory.class, OAuth2Utils.HTTP_TRANSPORT_FACTORY));
-    this.transportFactoryClassName = checkNotNull(this.transportFactory.getClass().getName());
-    this.audience = checkNotNull(audience);
-    this.subjectTokenType = checkNotNull(subjectTokenType);
-    this.tokenUrl = checkNotNull(tokenUrl);
-    this.credentialSource = checkNotNull(credentialSource);
-    this.tokenInfoUrl = tokenInfoUrl;
-    this.serviceAccountImpersonationUrl = serviceAccountImpersonationUrl;
-    this.quotaProjectId = quotaProjectId;
-    this.clientId = clientId;
-    this.clientSecret = clientSecret;
-    this.scopes =
-        (scopes == null || scopes.isEmpty()) ? Arrays.asList(CLOUD_PLATFORM_SCOPE) : scopes;
-    this.environmentProvider = SystemEnvironmentProvider.getInstance();
-
-    this.impersonatedCredentials = initializeImpersonatedCredentials();
+    this(
```

Comment:
In hindsight, we should have made these protected constructors accept only the `Builders` to avoid bloating all these constructor overloads.